core/src/main/java/org/elasticsearch/http/HttpServer.java

```diff
@@ -131,6 +137,22 @@ public void process(RestRequest request, RestChannel channel, RestFilterChain fi
         }
     }
 
+    void handleFavicon(HttpRequest request, HttpChannel channel) {
+        if (request.method() == RestRequest.Method.GET) {
+            try {
+                try (InputStream stream = getClass().getResourceAsStream("/config/favicon.ico")) {
```

Comment:
i dunno, it only impacts users using the browser. i figure that isnt perf critical, the server doesn't get hammered with those. maybe i'm wrong. file is 1KB.